pkg/services/librarypanels/librarypanels_test.go

```diff
@@ -28,6 +28,51 @@ func TestCreateLibraryPanel(t *testing.T) {
 			response = sc.service.createHandler(sc.reqContext, command)
 			require.Equal(t, 400, response.Status())
 		})
+
+	testScenario(t, "When an admin tries to create a library panel that does not exists, it should succeed",
+		func(t *testing.T, sc scenarioContext) {
+			command := getCreateCommand(1, "Text - Library Panel")
+			response := sc.service.createHandler(sc.reqContext, command)
+			require.Equal(t, 200, response.Status())
+
+			var result libraryPanelResult
+			err := json.Unmarshal(response.Body(), &result)
+			require.NoError(t, err)
+			var expected = libraryPanelResult{
+				Result: libraryPanel{
+					ID:       1,
+					OrgID:    1,
+					FolderID: 1,
+					UID:      result.Result.UID,
+					Name:     "Text - Library Panel",
+					Model: map[string]interface{}{
+						"datasource": "${DS_GDEV-TESTDATA}",
+						"id":         float64(1),
+						"title":      "Text - Library Panel",
+						"type":       "text",
+					},
+					Meta: LibraryPanelDTOMeta{
+						CanEdit:             true,
+						ConnectedDashboards: 0,
+						Created:             result.Result.Meta.Created,
+						Updated:             result.Result.Meta.Updated,
+						CreatedBy: LibraryPanelDTOMetaUser{
+							ID:        1,
+							Name:      "signed_in_user",
+							AvatarUrl: "/avatar/37524e1eb8b3e32850b57db0a19af93b",
+						},
+						UpdatedBy: LibraryPanelDTOMetaUser{
+							ID:        1,
+							Name:      "signed_in_user",
+							AvatarUrl: "/avatar/37524e1eb8b3e32850b57db0a19af93b",
+						},
+					},
+				},
+			}
```

Comment:
I'm thinking about extracting this out to it's own function. Is this something that I should do or is it more Go to keep it like this?